apps/af-mvp/src/pages/auth/sign-in/index.page.tsx

```diff
@@ -36,20 +36,22 @@ export default function SingInPage({
   const [isLoading, setIsLoading] = useState(true);
   const [isAuthenticated, setIsAuthenticated] = useState(false);
 
-  const checkAuthStatus = useCallback(async () => {
+  const checkAuthStatus = async () => {
     if (!isAuthenticated) {
       const isLoggedInCognito = (await fetchAuthIdToken()) !== null;
       setIsAuthenticated(isLoggedInCognito);
+      setIsLoading(false);
+    } else {
+      setIsLoading(false);
     }
-    setIsLoading(false);
-  }, [isAuthenticated, setIsLoading, setIsAuthenticated]);
+  };
 
+  // Set CSRF token, check auth status
+  // Disable eslint rule for this as we want to run this effect only once when the component mounts
   useEffect(() => {
-    if (isLoading) {
-      authStore.setCsrfToken(csrfToken);
-      checkAuthStatus();
-    }
-  }, [checkAuthStatus, isLoading, csrfToken]);
+    authStore.setCsrfToken(csrfToken);
+    checkAuthStatus();
+  }, []); // eslint-disable-line react-hooks/exhaustive-deps
```

Comment:
Tämä vaikuttaisi toimivan ilman varoitusta myös siten että vaan jättää dependency-lista-argumentin pois, mutta parempi näin merkattuna ja vielä kommentin kanssa niin viestittyy aije.